src/ORM/Association/BelongsToMany.php

```diff
@@ -1278,11 +1256,26 @@ protected function _buildQuery($options)
 
         // Ensure that association conditions are applied
         // and that the required keys are in the selected columns.
+
+        $tempName = $this->_name . '_CJoin';
```

Comment:
Is there anything special about this name or are you just trying to ensure uniqueness?